google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManager.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.apis;
+
+import com.google.api.services.servicemanagement.ServiceManagement;
+import com.google.api.services.servicemanagement.model.EnableServiceRequest;
+import com.google.cloud.tools.intellij.flags.PropertiesFileFlagReader;
+import com.google.cloud.tools.intellij.login.CredentialedUser;
+import com.google.cloud.tools.intellij.login.Services;
+import com.google.cloud.tools.intellij.project.CloudProject;
+import com.google.cloud.tools.intellij.resources.GoogleApiClientFactory;
+import com.google.cloud.tools.intellij.ui.GoogleCloudToolsIcons;
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.google.common.collect.Lists;
+import com.intellij.notification.NotificationDisplayType;
+import com.intellij.notification.NotificationGroup;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.diagnostic.Logger;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** Cloud API manager responsible for API management tasks on GCP such as API enablement. */
+class CloudApiManager {
+  private static final Logger LOG = Logger.getInstance(CloudApiManager.class);
+
+  private static final NotificationGroup NOTIFICATION_GROUP =
+      new NotificationGroup(
+          new PropertiesFileFlagReader().getFlagString("notifications.plugin.groupdisplayid"),
+          NotificationDisplayType.BALLOON,
+          true,
+          null,
+          GoogleCloudToolsIcons.CLOUD);
+  private static final String SERVICE_REQUEST_PROJECT_PATTERN = "project:%s";
+
+  private CloudApiManager() {}
+
+  /**
+   * Enables the supplied set of {@link CloudLibrary CloudLibraries} on GCP.
+   *
+   * @param libraries the set of {@link CloudLibrary CloudLibraries} to enable on GCP
+   * @param cloudProject the {@link CloudProject CloudProject} on which to enable the APIs
+   */
+  static void enableApis(Set<CloudLibrary> libraries, CloudProject cloudProject) {
+    Optional<CredentialedUser> user =
+        Services.getLoginService().getLoggedInUser(cloudProject.googleUsername());
+
+    if (!user.isPresent()) {
+      LOG.warn("Cannot enable APIs: logged in user not found.");
```

Comment:
This error is unlikely to happen, but maybe show this is in error balloon as well for consistency with other errors here?

Reply:
hmm. I actually don't think this error should ever happen unless something really goes wrong. I think changing it to log.error instead so that the user sees an ide plugin error and it gets reported to us makes sense. wdyt?

Reply:
sounds good, it does look like a hypothetical error only.

Reply:
changed